src/app/map-location/map-location-list/map-location-list.component.ts

```diff
@@ -219,4 +219,19 @@ export class MapLocationListComponent implements OnInit, OnChanges {
       });
     }
   }
+
+  onCalculateRoute() {
+    if (this.activeMapLocationId) {
+      const selectedLocation = this.mapLocations.find(location => location.id === this.activeMapLocationId);
+      if (selectedLocation) {
+        const destination = selectedLocation.coordinates;
+        this.mapService.mapLocationDetailsEventEmitter.emit(selectedLocation);
+      } else {
+        console.error('Selected location not found');
+      }
+    } else {
+      console.error('No active map location selected');
+    }
+  }
+
```

Comment:
nice to have:

I don't like nested if-else statements, because it is unreadable.
Consider something like this, without indentations, is easier to read:
```
onCalculateRoute() {
  if (!this.activeMapLocationId) {
    console.error('No active map location selected');
    return;
  }

  const selectedLocation = this.mapLocations.find(location => location.id === this.activeMapLocationId);
  if (!selectedLocation) {
    console.error('Selected location not found');
    return;
  }

  const destination = selectedLocation.coordinates;
  this.mapService.mapLocationDetailsEventEmitter.emit(selectedLocation);
}
```

Also, I am wondering if we really need as many log statements. these cases should be handled and showed to the user that he is doing something wrong. In the current state - only we as programmers  know that the user didn't select location or the map is not activated, but the user will not be opening the developer console

Reply:
@Drillllll what do you think, can we remove log statements from this review?

Reply:
I think we can leave it. Later, when snackbar is implemented we can utilize it to show user the corresponding error or warning messages